Src/ILGPU/Backends/SPIRV/ISPIRVBuilder.tt

```diff
@@ -0,0 +1,58 @@
+<#@ template hostspecific="true" language="C#" #>
+<#@ output extension=".cs"#>
+<#@ include file="BuilderCommon.ttinclude"
+#><#@ assembly name="System.Xml" #>
+<#@ assembly name="System.Core"#>
+<#@ import namespace="System.Linq"#>
+using System;
+using System.Collections.Generic;
+
+#nullable enable
+#pragma warning disable 1591
+
+namespace ILGPU.Backends.SPIRV {
+
+<#
+    PushIndent(standardIndent);
+#>
+/// <summary>
+/// Defines utility methods to generate SPIRV operations
+/// </summary>
+[CLSCompliant(false)]
+public interface ISPIRVBuilder
+{
+
+    byte[] ToByteArray();
+
+    void AddMetadata(uint magic, uint version, uint genMagic, uint bound, uint schema);
+
+    // This is the best way I could come up with to
+    // handle trying to merge different builders
+    // Implementing classes will kinda just have to
+    // deal with it
+    void Merge(ISPIRVBuilder other);
+
+<#
+    PushIndent(standardIndent);
+    foreach (var instruction in grammar.Instructions.Instructions)
+    {
+        if (!IsInstructionCLSCompliant(instruction))
+        {
+#>[CLSCompliant(false)]
```

Comment:
Remark: if we change the interface to work on `SpirVInstructions`, we can skip all of these attributes.